spek-table-driven-extension/src/main/kotlin/org/jetbrains/spek/table/TableDriven.kt

```diff
@@ -0,0 +1,384 @@
+package org.jetbrains.spek.table
+
+import org.jetbrains.spek.api.dsl.Pending
+import org.jetbrains.spek.api.dsl.Spec
+import org.jetbrains.spek.api.dsl.TestBody
+import org.jetbrains.spek.api.dsl.TestContainer
+import org.jetbrains.spek.meta.Experimental
+
+data class Example1<out P1> internal constructor(val p1: P1)
+
+@Experimental
+fun <P1> example(p1: P1): Example1<P1> = Example1(p1)
+
+@Experimental
+fun <P1> TestContainer.with(vararg examples: Example1<P1>, testContent: TestContainer. (P1) -> Unit) {
+    examples.forEach {
+        val proxyContainer = object : TestContainer by this@with {
+            override fun test(description: String, pending: Pending, body: TestBody.() -> Unit) {
+                this@with.test("$description ($it)",pending,body)
+            }
+        }
+        testContent.invoke(proxyContainer, it.p1)
+    }
+}
+
+
+data class Example2<out P1, out P2> internal constructor(val p1: P1, val p2: P2)
+
+@Experimental
+fun <P1, P2> example(p1: P1, p2: P2): Example2<P1, P2> = Example2(p1, p2)
+
+@Experimental
+fun <P1, P2> TestContainer.with(vararg examples: Example2<P1, P2>, testContent: TestContainer. (P1, P2) -> Unit) {
```

Comment:
I like the `forEvery` version because it will neatly be enhance by property based testing. We'll get something like : 
```kotlin
forEvery(Int.class) { n ->
...
}
```
If we want two different APIs, `givenData` is better but sounds awkward to me :(